### ISSUE TITLE
Out of Bounds Error assigning station latitude to SAC Header

### ISSUE THREAD

**Issue by schaefferaj** [obspy/obspy#1300]
In obspy v1.0.0, SACTrace  can be used to initialize the sac header dictionary to get around some changes in the SAC io modules resulting in another bug (see [https://github.com/obspy/obspy/issues/1285] for details).

After reading a miniseed file to a stream or downloading data to a stream via a client, the trace is converted via SACTrace:
`trsac = obspy.io.sac.sactrace.SACTrace.from_obspy_trace(tr)`

The station latitude can now be assigned:
`trsac.stla = -35.0`
However, this produces the following error:
`Latitude of Point 1 out of bounds! (-90 <= lat1 <=90)`

The value is clearly in the appropriate range of values. Furthermore, in ipython, querying the value shows it was actually applied:
`print trsac.stla`
`    -35.0   `
It appears that the value is being assigned, but following the assignment of this single value, gps2_distazimuth in obspy.geodetics is being called, which is where the error message is originating.

This is on a computer running Ubuntu 14.04 x86_64 with python2.7 and obspy v1.0.0 installed via the package manager.
Note that no error is produced running the same operation on the same data on a 64bit Windows 10 Laptop running python3.4 and obspy v1.0.0.

**Comment by megies** [comment]
I can confirm:
```python
from obspy import read
from obspy.io.sac.sactrace import SACTrace
st = read()
trsac = SACTrace.from_obspy_trace(st[0])
trsac.stla = -35.0
```

```
Traceback (most recent call last):
  File "/tmp/sacbug.py", line 5, in <module>
    trsac.stla = -35.0
  File "/home/megies/git/obspy/obspy/io/sac/sactrace.py", line 549, in set_geo
    self._set_distances()
  File "/home/megies/git/obspy/obspy/io/sac/sactrace.py", line 1547, in _set_distances
    self.stlo)
  File "/home/megies/git/obspy/obspy/geodetics/base.py", line 248, in gps2dist_azimuth
    raise ValueError(msg)
ValueError: Latitude of Point 1 out of bounds! (-90 <= lat1 <=90)
```
`gps2dist_azimuth` gets called with three `None` values. There should be a check if all four necessary values (event lon/lat + station lon/lat) are set before trying to get a distance for the header. @jkmacc-LANL, I didn't immediately see where it's best to do this `if` check, maybe you can add it where it fits best?

**Comment by jkmacc-LANL** [comment]
I'll take care of it  Thanks for noticing this.

**Comment by megies** [comment]
> EDIT: should I make a separate branch and PR for this issue, or can I sneak a commit in on #1285 ?

If you prefer to do it over in that other branch.. fine with me. (Adding `fixes #1300` in the respective commit message will automatically close this one, as soon as it appears in master)

**Comment by jkmacc-LANL** [comment]
Fixed in #1285
